### PR TITLE
Adopt AIP-200: Precedent

### PR DESCRIPTION
**Pull request by rofrankel**

### CODE REVIEW

**Review by earth2marsh — APPROVED** [review]
LGTM.

One small q, the [original AIP](https://github.com/aip-dev/google.aip.dev/blob/master/aip/general/0200.md?plain=1#L146) seems to indicate a connection to other AIPs:
```[aip-8]: ./0008.md
[aip-142]: ./0142.md```
Are those irrelevant for AEPs? If so, carry on!

File: aep/general/0200/aep.md.j2 [hunk 1]
```diff
@@ -1,5 +1,140 @@
...
+  // Output only. Duration of the time window, automatically chosen to be
+  // smallest possible in the given scenario.
+  // (-- aep.dev/not-precedent: This preceded the AEP standards.
+  //     Ordinarily, this type should be `google.protobuf.Duration`. --)
+  string duration = 3;
+}
+```
+{% tab oas %}
+
+**Note:** OAS example not yet written.
```
Comment (earth2marsh) [inline aep/general/0200/aep.md.j2:52]: Is there an issue to track this as needing OpenAPI?

File: aep/general/0200/aep.md.j2 [hunk 2]
```diff
@@ -1,5 +1,140 @@
...
+The comment should also include an explanation of what violates standards and
+why it is necessary. For example:
+
+{% tab proto %}
+
+```proto
+message DailyMaintenanceWindow {
+  // Time within the maintenance window to start the maintenance operations.
+  // It must use the format "HH MM", where HH : [00-23] and MM : [00-59] GMT.
+  // (-- aep.dev/not-precedent: This was designed for consistency with crontab,
```
Comment (toumorokoshi) [inline aep/general/0200/aep.md.j2:38]: we should make `aep.dev/not-precedent` redirect to this page.

File: aep/general/0200/aep.md.j2 [hunk 3]
```diff
@@ -1,5 +1,140 @@
...
+}
+```
+{% tab oas %}
+
+**Note:** OAS example not yet written.
+
+{% endtabs %}
+
+**Important:** APIs should only be considered to be precedent-setting if they
+are in beta or GA.
```
Comment (toumorokoshi) [inline aep/general/0200/aep.md.j2:57]:
```suggestion
are in beta or stable.
```

weird nomenclature distinction here: 181 calls the concept of GA stable: https://google.aip.dev/181#stable

**Review by toumorokoshi — COMMENTED** [review]
Although I like the idea in spirit of outlining what situations we should allow contradictions to the AEPs, I feel like the addition of this ends up with a lot of holes in the AEPs that can now be poked. Basically using the various reasons outlined below (e.g. technical implementation) as a justification to violate the spec.

I feel like it's worth revisiting some big chunks of this from the perspective of violating a spec - e.g. what valid reasons are there for an API representation that claims it's openapi compatible to have exceptions in adhering to the spec?

if an API does violate the spec - do we still call it AEP compliant at that point?

**Comment by rofrankel** [comment]
> LGTM.
> 
> One small q, the [original AIP](https://github.com/aip-dev/google.aip.dev/blob/master/aip/general/0200.md?plain=1#L146) seems to indicate a connection to other AIPs:
> 
> ```
> [aip-8]: ./0008.md
> [aip-142]: ./0142.md
> ```
> 
> Are those irrelevant for AEPs? If so, carry on!

The AIP-8 link was unused, and I replaced the 142 link with just `AEP-142`, which the site generator linkifies without needing a Markdown link.

File: aep/general/0200/aep.md.j2 [hunk 4]
```diff
@@ -1,5 +1,140 @@
 # Precedent
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/34) for more
-information.
+Many times, APIs are written in ways that do not match new guidance that is
+added to these standards after those APIs have already been released.
+Additionally, sometimes it can make sense to intentionally violate standards
+for particular reasons, such as maintaining consistency with established
+systems, meeting stringent performance requirements, or other practical
```
[the remark this answers is not in the document]
Comment (rofrankel) [reply aep/general/0200/aep.md.j2:7]: I agree with the spirit of your comment, at least. 

But I don't think a list of AIPs it's okay to violate in this case is the right approach.  And I'm not quite sure what else to write.  Any ideas?  Practically speaking, sometimes you have to write a new API for an old system.

(A real example from Roblox:  we had an existing service (originally only accessed via a Lua library provided to game engine scripts) that allows `/` in resource IDs.  This is obviously problematic for a resource-oriented HTTP API.  And yes, there are workarounds such as URL encoding, but ultimately allowing `/` in resource IDs is not something you'd ever want to do for a new service if following AEPs.)

File: aep/general/0200/aep.md.j2 (hunk 2 above)
Comment (rofrankel) [reply aep/general/0200/aep.md.j2:38]: Done.

File: aep/general/0200/aep.md.j2 (hunk 1 above)
Comment (earth2marsh) [reply aep/general/0200/aep.md.j2:52]: SGTM

File: aep/general/0200/aep.md.j2 (hunk 4 above)
[the remark this answers is not in the document]
Comment (earth2marsh) [reply aep/general/0200/aep.md.j2:7]: I added this as a discussion topic for Fri, as it seems like a good topic for further discussion. If we've resolved it by then, great!

File: aep/general/0200/aep.md.j2 (hunk 1 above)
Comment (toumorokoshi) [reply aep/general/0200/aep.md.j2:52]: done: https://github.com/aep-dev/aep.dev/issues/156

**Review by mkistler — APPROVED** [review]
Looks good. 👍

**Pull request merged**